org.mwc.debrief.legacy/src/Debrief/ReaderWriter/BRT/BRTImporter.java

```diff
@@ -0,0 +1,277 @@
+ *    modify it under the terms of the Eclipse Public License v1.0
+ *    (http://www.eclipse.org/legal/epl-v10.html)
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ */
+package Debrief.ReaderWriter.BRT;
+
+import java.awt.Color;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+
+import Debrief.Wrappers.SensorWrapper;
+import Debrief.Wrappers.TrackWrapper;
+import Debrief.Wrappers.Track.LightweightTrackWrapper;
+import MWC.GUI.Editable;
+import MWC.GUI.Layer;
+import MWC.GUI.Layers;
+import MWC.GUI.Tools.Action;
+import MWC.TacticalData.NarrativeWrapper;
+import junit.framework.TestCase;
+
+public class BRTImporter
+{
+
+  /**
+   * layer that we are going to be load data to
+   * 
+   */
+  private final Layers _layers;
+
+  /**
+   * brt info given by user.
+   */
+  private final BRTHelper _brtHelper;
+
+  private BRTData brtData;
+
+  /**
+   * Default Constructor
+   * 
+   * @param hB
+   *          Contains the information given by the user using the UI.
+   * @param _layers
+   *          layers available in the plot
+   */
+  public BRTImporter(BRTHelper hB, Layers _layers)
+  {
+    super();
+    this._layers = _layers;
+    this._brtHelper = hB;
+  }
+
+  public BRTData getBrtData()
+  {
+    return brtData;
+  }
+
+  /**
+   * loop through layers, find tracks
+   * 
+   * @param layers
+   *          Layers to loop
+   * @return Tracks available in the layers
+   */
+  private TrackWrapper[] getTracks()
+  {
+    List<TrackWrapper> result = new ArrayList<>();
+
+    Enumeration<Editable> layerIterator = _layers.elements();
+
+    while (layerIterator.hasMoreElements())
+    {
+      Layer layer = (Layer) layerIterator.nextElement();
+      if (TrackWrapper.class.isAssignableFrom(layer.getClass()))
+      {
+        result.add((TrackWrapper) layer);
+      }
+    }
+
+    return result.toArray(new TrackWrapper[]
+    {});
+  }
+
+  public ImportBRTAction importThis(final String fName, final InputStream is)
+  {
+    // TODO.
+    return createImportAction(fName);
+  }
+
+  private ImportBRTAction createImportAction(final String fName)
+  {
+    // find track
+    TrackWrapper track = findTrack();
+    SensorWrapper sensor = new SensorWrapper(fName);
+    // load data
+    // determine color/cut-length to sensor
+    // create sensor
+    final ImportBRTAction action = new ImportBRTAction(sensor, track);
+    // set default color
+    // set array offset (if needed)
+    //loadThese(Sensor…, color, cut-length)
+    return action;
+  }
+
+  private TrackWrapper findTrack()
+  {
+    // TODO
+    return null;
+  }
+
+  public static class ImportBRTAction implements Action
+  {
```

Comment:
@saulhidalgoaular - I'm being picky here, but...

For these two member fields, we know they aren't going to change.  So we should mark them as final, and then the compiler will _look after us_ by ensuring the never get changed.  It's the same for their declaration as parameters in the constructor.

We do have the Debrief code-cleanup settings.  With these loaded, when you run **Source > Clean up**, it will apply final when it's obvious they don't get changed.

But, in my opinion doing it early means the compile is _looking after you_ while you're still developing it.

(more on the settings here: https://github.com/debrief/debrief/wiki/On-boarding#project-settings )

Reply:
oh, I see. I just changed it!